addons/test_mass_mailing/tests/test_blacklist.py

```diff
@@ -8,15 +8,83 @@
 
 class TestBLAccessRights(common.MassMailingCase):
 
+    def setUp(self):
+        super(TestBLAccessRights, self).setUp()
+        self.bl_rec = self.env['mail.blacklist'].create([
+            {'email': 'Not A Stark <john.snow@example.com>'},
+        ])
+
+        self.user_portal = self.env['res.users'].with_context(self._quick_create_user_ctx).create({
```

Comment:
Unrelated to the PR, but seeing the number of times we have this pattern in our tests makes me wonder why we do not create a helper method `_new_test_user()` which that takes optional parameters for overriding `groups` and basic `vals`, with the rest auto-generated. Seems it could minimize this boilerplate code, to e.g.:
```python
self.user_portal = self.__new_test_user(login='port', groups='base.group_portal')
```